src/qubx/core/context.py

```diff
@@ -690,8 +706,19 @@ def trade(
         type = "limit" if price is not None else "market"
         logger.debug(f"(StrategyContext) sending {type} {side} for {size_adj} of {instrument.symbol} ...")
         client_id = self._generate_order_client_id(instrument.symbol)
+
+        order_params = {}
+        if self.broker_provider.is_simulated_trading:
+            if "fill_at_price" in kwargs and price is not None:
+                fill_at_price = kwargs["fill_at_price"]
+                order_params["fill_at_price"] = fill_at_price
+                if fill_at_price:
+                    # assume worse case
+                    # TODO: add an additional flag besides price to indicate order type
+                    type = "market"
+
         order = self.trading_service.send_order(
-            instrument, side, type, size_adj, price, time_in_force=time_in_force, client_id=client_id
+            instrument, side, type, size_adj, price, time_in_force=time_in_force, client_id=client_id, **order_params
```

Comment:
May be we could just pass **kwargs as is to send_order and avoid checking at 711 : 719 ?

Reply:
yes-yes, makes sense